expression/builtin_time.go

```diff
@@ -746,3 +746,34 @@ func parseDayInterval(sc *variable.StatementContext, value types.Datum) (int64,
 	}
 	return value.ToInt64(sc)
 }
+
+// https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_timestampdiff
+func builtinTimestampDiff(args []types.Datum, ctx context.Context) (d types.Datum, err error) {
+	sc := ctx.GetSessionVars().StmtCtx
+	t1, err := convertDatumToTime(sc, args[1])
+	if err != nil {
+		return d, errorOrWarning(err, ctx)
+	}
+	t2, err := convertDatumToTime(sc, args[2])
+	if err != nil {
+		return d, errorOrWarning(err, ctx)
+	}
+	if t1.Time.Month() == 0 || t1.Time.Day() == 0 || t2.Time.Month() == 0 || t2.Time.Day() == 0 {
+		return d, errorOrWarning(types.ErrInvalidTimeFormat, ctx)
+	}
+
+	v := types.TimestampDiff(args[0].GetString(), t1, t2)
+	d.SetInt64(v)
+	return
+}
+
+// errorOrWarning reports error or warning depend on the context.
+func errorOrWarning(err error, ctx context.Context) error {
```

Comment:
No other place use it currently, and we don't fully support warning now.
Maybe it would be use in the future, but we can refact then.